pkg/bgp/speaker/speaker.go

```diff
@@ -374,6 +375,103 @@ func (s *MetalLBSpeaker) OnDeleteNode(node *v1.Node) error {
 	return nil
 }
 
+// OnAddCiliumNode notifies the Speaker of a new CiliumNode.
+func (s *MetalLBSpeaker) OnAddCiliumNode(node *ciliumv2.CiliumNode) error {
+	if s.shutDown() {
+		return ErrShutDown
+	}
+	if node.GetName() != nodetypes.GetName() {
+		return nil // We don't care for other nodes.
+	}
+	var (
```

Comment:
Since this code is essentially a copy of the K8s Node version, we should probably de-dup this by creating a new function which is agnostic to the type of the input object (CN or K8s Node) and just works on raw fields like meta & CIDRs. We still need to keep the CiliumNode handlers, but they should just call down to a function like below:

```
// feel free to use better names than I did; this is half real code, half pseudo.
func (s *MetalLBSpeaker) reconcileAdd(meta types.ObjectMeta, cidr CIDRType) error {
	if s.shutDown() {
		return ErrShutDown
	}
	if meta.GetName() != nodetypes.GetName() {
		return nil // We don't care for other nodes.
	}
	...
```

Reply:
Good feedback, thanks! I thought about this as well, but didn't realize everything besides the CIDRs relied on the `ObjectMeta`. Addressed, PTAL.